dkg/pedersen/handler.go

```diff
@@ -530,3 +607,299 @@ func (h *Handler) announceDkgPublicKey(out mino.Sender, from mino.Address) error
 
 	return nil
 }
+
+// handles the resharing request - acts differently for the new and old and common nodes
+func (h *Handler) handleReshare(out mino.Sender,
+	from mino.Address, msg types.ResharingRequest, done chan struct{}) error {
+	// if the node is completely new
+	if h.startRes.distrKey == nil {
+		if h.startRes.Done() {
+			dela.Logger.Warn().Msgf(
+				"%v ignored reshare request from %v as it has"+
+					" started before\n", h.me, from)
+			return xerrors.Errorf("dkg is started")
+		}
+
+	} else { // if the node is in the old committee or is a common node
+		if !h.startRes.Done() {
+			dela.Logger.Warn().Msgf(
+				"%v ignored reshare request from %v as it is not"+
+					" started yet\n", h.me, from)
+			return xerrors.Errorf("dkg is not started")
+		}
+	}
+
+	if h.startRes.IsReshare() {
```

Comment:
a point @nkcr raised in my PR, is that because we're only taking a lock within each `h.startRes` function, and only doing one check at a time, we might actually be exposed to concurrency issues.

We really should lock the whole state from the beginning of this function to line 640, and only release it then.

I'm saying this FYI, but I'm happy to fix it for the both of us in my own PR

Reply:
yes, let's fix that with your PR @pierluca